client/landing/stepper/declarative-flow/internals/components/step-route/hooks/use-step-route-tracking/index.tsx

```diff
@@ -50,7 +68,7 @@ export const useStepRouteTracking = ( {
 	useEffect( () => {
 		// We record the event only when the step is not empty. Additionally, we should not fire this event whenever the intent is changed
 		if ( ! hasRequestedSelectedSite || skipTracking ) {
```

Comment:
I am a bit confused about the logic of the cleanup function (return from use effect hook).
Ideally, cleanup functions should not be conditional.
Furthermore, I noticed that [here](https://github.com/Automattic/wp-calypso/pull/94528/files#diff-1ceaa1de69b40e0a07953f76572ba4484bd63b716ce1f4ff4e01b6015d4eb5c9R105) the same return is happening, so the condition looks unnecessary.

Reply:
Yes, you are correct. This needs to go (it's on my list). 

My question is whether we need to early return at all i.e. why a step that renders shouldn't be logged (assuming I'm not missing something in how routes are handled). Any chance you can help confirm?

Reply:
^ from comment above https://github.com/Automattic/wp-calypso/pull/94528#discussion_r1758469725

Reply:
I cleaned up / removed the call. Let's continue discussing it here pls. 

Why do we not log/record a step that's made it into a `StepRoute`? which reflects on why we `shouldSkipRender` in `StepRoute` (is it anything that lingered from before migrating the `User` step to Stepper?)

Reply:
> Why do we not log/record a step that's made it into a StepRoute? which reflects on why we shouldSkipRender in StepRoute (is it anything that lingered from before migrating the User step to Stepper?)

Could you elaborate a bit more on this? I didn't get what you meant.

Reply:
I'm trying to justify/understand the presence of `skipTracking` i.e. whether we should skip tracking a step if the user has accessed it and it is a valid step route. 

Unless I am missing something, this happens when user accesses a given step and it requires auth (e.g. going to `/setup/onboarding/plans` while logged out), at which point the user is sent to auth route/step. Should we not track `calypso_signup_step_start` on the plans step in that case? 🤔

Curious if we lose something from analytics when it happens. I guess also a question for @travisw / @skim1220 ?

Reply:
It seems that, based on the code [linked here](https://github.com/Automattic/wp-calypso/blob/update/stepper-tracking-complete-step/client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx#L28), this is designed to prevent tracking events when the step either requires user authentication or isn't rendered at all.

The rationale for not logging events in these cases is likely because the user will be redirected elsewhere, likely to an authentication page.

Perhaps @escapemanuele can provide more context on this?

Reply:
> Unless I am missing something, this happens when user accesses a given step and it requires auth (e.g. going to /setup/onboarding/plans while logged out), at which point the user is sent to auth route/step. Should we not track calypso_signup_step_start on the plans step in that case?

If a user tries to access the plans step while logged-out and is redirected to the user steps, then I'd intuitively expect the plans step not to fire. Then once the user logs in and is redirected back to the plans step and they actually see the step, then it should fire.

Typically I think these types of events are meant to track the user's experience rather than the code's experience.

also cc @MicBosi

Reply:
Agree for the most part, but also think that it's through the system logs that you can navigate the full extend of their experience. Stepper being user-facing framework, pretty much whatever it does is typically a reaction.

The initial request is also part of the user experience, no? So to understand their journey I'd be looking at the full spectrum below:

1. User goes to `/setup/onboarding/plans`
2. Redirected to `/setup/onboarding/user`
3. Signs up
4. Rdirected to `/setup/onboarding/domains` (not plans)

Ultimately not accounting anything around 1 may be a data point missing. Insignificant as it may be/seem.

Reply:
> Ultimately not accounting anything around 1 may be a data point missing. Insignificant as it may be/seem.

That's a good point. It could be nice to have that data if needed.

Reply:
> That's a good point. It could be nice to have that data if needed.

Cool. moved this to an issue to follow up on https://github.com/Automattic/wp-calypso/issues/94715